core/primitives/src/crypto.rs

```diff
@@ -307,18 +309,23 @@ pub trait Pair: Sized + 'static {
 	///
 	/// This is only for ephemeral keys really, since you won't have access to the secret key
 	/// for storage. If you want a persistent key pair, use `generate_with_phrase` instead.
-	fn generate() -> Self;
+	fn generate() -> (Self, Self::Seed) {
+		let mut csprng: OsRng = OsRng::new().expect("OS random generator works; qed");
+		let mut seed = Self::Seed::default();
+		csprng.fill_bytes(seed.as_mut());
```

Comment:
Not sure about this, but maybe still noteworthy.
Here is the quote from the [fill_bytes](https://rust-random.github.io/rand/rand_core/trait.RngCore.html#tymethod.fill_bytes) docs (emphasis is mine)

> RNGs must implement at least one method from this trait directly. In the case this method is not implemented directly, it can be implemented via next_u* or via try_fill_bytes; if this generator can fail the implementation must choose how best to handle errors here (e.g. **panic with a descriptive message** or **log a warning and retry a few times**).
> 
> This method should guarantee that dest is entirely filled with new data, and **may panic** if this is impossible (e.g. reading past the end of a file that is being used as the source of randomness).

[And](https://rust-random.github.io/rand/rand_os/struct.OsRng.html#blocking-and-error-handling)

> It is possible that when used during early boot the first call to OsRng will block until the system's RNG is initialised. It is also possible (though highly unlikely) for OsRng to fail on some platforms, most likely due to system mis-configuration.
>
> After the first successful call, it is highly unlikely that failures or significant delays will occur (although performance should be expected to be much slower than a user-space PRNG).

[And](https://docs.rs/getrandom/0.1.3/getrandom/#early-boot)

> Some operating systems always block the thread until the RNG is securely seeded. This can take anywhere from a few seconds to **more than a minute**. Others make a best effort to use a seed from before the shutdown and don't document much.

[And](https://docs.rs/getrandom/0.1.3/getrandom/#error-handling)

> We always choose failure over returning insecure "random" bytes. In general, on supported platforms, failure is highly unlikely, **though not impossible**. If an error does occur, then it is likely that it will occur on every call to getrandom, hence after the first successful call one can be reasonably confident that no errors will occur.

Reply:
that stuff only happens in misconfigured systems. and i really don't care about misconfigured systems.